common/djangoapps/course_modes/views.py

```diff
@@ -135,6 +136,7 @@ def get(self, request, course_id, error=None):
 
         return render_to_response("course_modes/choose.html", context)
 
+    @method_decorator(transaction.non_atomic_requests)
     @method_decorator(login_required)
```

Comment:
If we want it execute before any transaction setup, wouldn't we want it to be the outermost one? However, given the logic in https://github.com/django/django/blob/1.8.5/django/core/handlers/base.py#L129-L132, I think `transaction.non_atomic_request` must be at the outermost decorator.